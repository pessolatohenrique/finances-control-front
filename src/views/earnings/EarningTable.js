import React from "react";
import moment from "moment";
import {
  TableContainer,
  Table,
  TableHead,
  TableBody,
  TableCell,
  TableRow,
} from "@mui/material";
import { StyledTableCell } from "../../constants/CustomStyles";

export function EarningTable({ budget }) {
  return (
    <TableContainer>
      <Table>
        <TableHead>
          <TableRow>
            <StyledTableCell>Descrição</StyledTableCell>
            <StyledTableCell>Valor</StyledTableCell>
            <StyledTableCell>Data</StyledTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {budget?.Earnings.map((row) => {
            return (
              <TableRow key={row.id} data-testid="row-table">
                <TableCell>{row.name}</TableCell>
                <TableCell>{row.UserEarning.value}</TableCell>
                <TableCell>
                  {moment(row.UserEarning.transaction_date).format(
                    "DD/MM/YYYY"
                  )}
                </TableCell>
              </TableRow>
            );
          })}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
