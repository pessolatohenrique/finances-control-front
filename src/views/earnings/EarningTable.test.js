import moment from "moment";
import { render, screen } from "@testing-library/react";
import { EarningTable } from "./EarningTable";

const budget = {
  Earnings: [
    {
      id: 1,
      name: "SalÃ¡rio",
      UserEarning: {
        value: 1000,
        transaction_date: moment().format("YYYY-MM-DD"),
      },
    },
    {
      id: 2,
      name: "Dividendos",
      UserEarning: {
        value: 500,
        transaction_date: moment().format("YYYY-MM-DD"),
      },
    },
  ],
};

describe("Earning Table", () => {
  it("should generate snapshot", () => {
    const { container } = render(<EarningTable budget={budget} />);
    expect(container).toMatchSnapshot();
  });

  it("should generate table", () => {
    render(<EarningTable budget={budget} />);
    const rowTable = screen.getAllByTestId("row-table");
    expect(rowTable.length).toBe(2);
  });
});
