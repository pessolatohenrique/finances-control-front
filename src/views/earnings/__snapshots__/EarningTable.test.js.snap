// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Earning Table should generate snapshot 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Descrição
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Valor
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1w4f6gj-MuiTableCell-root"
            scope="col"
          >
            Data
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          data-testid="row-table"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Salário
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            1000
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            17/05/2022
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          data-testid="row-table"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Dividendos
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            500
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            17/05/2022
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
